action/jsonapi.go

```diff
@@ -107,13 +113,43 @@ func respondMessage(ctx context.Context, s *Action, msgBytes []byte) error {
 	switch message.Type {
 	case "query":
 		return respondQuery(s, msgBytes)
+	case "queryHost":
+		return respondHostQuery(s, msgBytes)
 	case "getLogin":
 		return respondGetLogin(ctx, s, msgBytes)
 	default:
 		return fmt.Errorf("Unknown message of type %s", message.Type)
 	}
 }
 
+func respondHostQuery(s *Action, msgBytes []byte) error {
+	var message queryHostMessage
+	if err := json.Unmarshal(msgBytes, &message); err != nil {
+		return err
+	}
+
+	l, err := s.Store.List(0)
+	if err != nil {
+		return err
+	}
+	choices := make([]string, 0, 10)
+
+	reQuery := fmt.Sprintf("(^|.*/)%s($|/.*)", regexSafeLower(message.Host))
```

Comment:
This would definitely warrant some in-place documentation.

Reply:
Comment added.